runtime/sema/type.go

```diff
@@ -9534,3 +9552,34 @@ func init() {
 		})
 	}
 }
+
+var NativeInterfaceTypes = map[string]*InterfaceType{}
+
+func init() {
+	interfaceTypes := []*InterfaceType{
+		StringerType,
+	}
+
+	for len(interfaceTypes) > 0 {
+		lastIndex := len(interfaceTypes) - 1
+		interfaceType := interfaceTypes[lastIndex]
+		interfaceTypes[lastIndex] = nil
+		interfaceTypes = interfaceTypes[:lastIndex]
+
+		NativeInterfaceTypes[interfaceType.QualifiedIdentifier()] = interfaceType
+
+		nestedTypes := interfaceType.NestedTypes
+		if nestedTypes == nil {
+			continue
+		}
+
+		nestedTypes.Foreach(func(_ string, nestedType Type) {
+			nestedInterfaceType, ok := nestedType.(*InterfaceType)
+			if !ok {
+				return
+			}
+
+			interfaceTypes = append(interfaceTypes, nestedInterfaceType)
+		})
+	}
```

Comment:
It might be good to refactor this into a function, and also use it in the `init` block initializing `NativeCompositeTypes`